server/handler.go

```diff
@@ -988,6 +1008,15 @@ func (h *jobsGetQueryResultsHandler) Handle(ctx context.Context, r *jobsGetQuery
 		return nil, err
 	}
 	rows := internaltypes.Format(response.Schema, response.Rows, r.useInt64Timestamp)
+
+	if r.maxResults == 0 {
+		rows = nil
+	}
+
+	if r.maxResults != maxResultsDefaultValue {
+		rows = rows[:min(int64(len(rows)), r.maxResults)]
```

Comment:
If `-1` is set as `maxResults`, I think panic may occur here

Reply:
-1 is the `maxResultsDefaultValue`. I can add a validation for < -1